mutation/mutation_partition.cc

```diff
@@ -462,6 +449,25 @@ stop_iteration mutation_partition::apply_monotonically(const schema& s, mutation
           throw;
       }
     }
+
+    // Compaction is attempted only in preemptible contexts because it can be expensive to perform and is not
+    // necessary for correctness.
+    if (p._tombstone && preemptible) {
+        // p._tombstone is already applied to _tombstone
+        rows_type::iterator i;
+        if (res._stage == apply_resume::stage::partition_tombstone_compaction) {
+            i = _rows.lower_bound(res._pos, cmp);
+        } else {
+            i = _rows.begin();
+        }
+        if (apply_tombstone_to_rows(apply_resume::stage::partition_tombstone_compaction,
+                                               _tombstone, i, _rows.end()) == stop_iteration::no) {
+            return stop_iteration::no;
+        }
+        // TODO: Drop redundant range tombstones
+        p._tombstone = {};
+    }
+
```

Comment:
Why this code movement?

Reply:
sorry, this was leftover from debugging as well.
although it might merit an independent change (that should be considered on its own).
the reason is to make compaction more effective.
the current code applies the tombstone asymmetrically, but we may want to compact the result of the applied partition.